libsrc/osic1p/cgetc.c

```diff
@@ -0,0 +1,167 @@
+/*
+ * Input routines for OSI C1P platform.
+ */
+
+#include <conio.h>
+#include <ctype.h>
+#include <peekpoke.h>
+
+// Address of hardware keyboard register.
+#define KEYBOARD 0xdf00
+
+// Some special keys.
+#define ESC 0x1b
+#define CR 0x0d
+#define LF 0x0a
+
+// Lookup table of keys, given keyboard row and bit numbers.
+const unsigned char keys[8][8] = {
+    { 0,   0,   0,    0,    0,  ESC,   0,  0  }, // Row 0
+    { 0, 'p', ';',  '/',  ' ',  'z', 'a', 'q' }, // Row 1
+    { 0, ',', 'm',  'n',  'b',  'v', 'c', 'x' }, // Row 2
+    { 0, 'k', 'j',  'h',  'g',  'f', 'd', 's' }, // Row 3
+    { 0, 'i', 'u',  'y',  't',  'r', 'e', 'w' }, // Row 4
+    { 0,   0,   0,   CR,   LF,  'o', 'l', '.' }, // Row 5
+    { 0,   0, '_',  '-',  ':',  '0', '9', '8' }, // Row 6
+    { 0, '7', '6',  '5',  '4',  '3', '2', '1' }, // Row 7
+};
+
+// Table of bitmasks for active keys in each row.
+const unsigned char columnMask[8] = {
+    0x20, // Row 0, only ESC key
+    0xfe, // Row 1
+    0xfe, // Row 2
+    0xfe, // Row 3
+    0xfe, // Row 4
+    0xf8, // Row 5
+    0xfc, // Row 6
+    0xfe, // Row 7
+};
+
+// Bit masks for modifier keys.
+const unsigned char controlMask = 0x40; // Control
+const unsigned char shiftMask = 0x06;   // Left Shift, Right Shift
+const unsigned char lockMask = 0x01;    // Shift Lock
```

Comment:
3 x static

Reply:
Scalar values that never change should be used as either literals, macroes, or enumerations.  Using a variable to hold a constant makes cc65 generate inefficient code.